File: src/main/scala/SparkTPApp1.scala
import org.apache.spark.{SparkConf}
import org.apache.spark.sql.SparkSession

object SparkTPApp1 {

  val compte = "ecoquery"

  def compteAB(logFile: String, sc: SparkSession): (Long, Long) = {
    val logData = sc.read.textFile(logFile).cache()
    val numAs = logData.filter(line => line.contains("a")).count()
    val numBs = logData.filter(line => line.contains("b")).count()
    (numAs, numBs)
  }

  def main(args: Array[String]): Unit = {
    val logFile = "hdfs:///user/" + compte + "/README.md"
    val sc = SparkSession.builder.appName("SparkTPApp1-" + compte).getOrCreate()
    val (nbA, nbB) = compteAB(logFile, sc)
    println("\n\nLines with a: %s, Lines with b: %s\n\n".format(nbA, nbB))
  }

}

File: python/SparkTPApp1.py
# -*- coding: utf-8 -*-
import sys


compte = "ecoquery"

def compteAB(logFile, sc):
    logData = sc.read.text(logFile).cache()
    numAs = logData.filter(logData.value.contains("a")).count()
    numBs = logData.filter(logData.value.contains("b")).count()
    return (numAs, numBs)

def getSpark(appName):
    from pyspark.sql import SparkSession
    return SparkSession.builder.appName(appName).getOrCreate()

if __name__ == "__main__":
    logFile = "hdfs:///user/" + compte + "/README.md"
    appName = "SparkTPApp1-" + compte
    spark = getSpark(appName)
    (nbA, nbB) = compteAB(logFile, spark)
    print("\n\nLines with a: {}, Lines with b: {}\n\n".format(nbA, nbB))
    spark.stop()
